cip.go

```diff
@@ -108,38 +112,68 @@ func main() {
 	var mfest reg.Manifest
 	var srcRegistry *reg.RegistryContext
 	var err error
+	var mfests []reg.Manifest
+	sc := reg.SyncContext{}
+	mi := make(reg.MasterInventory)
+
 	if len(*snapshotPtr) > 0 {
 		srcRegistry = &reg.RegistryContext{
 			Name:           reg.RegistryName(*snapshotPtr),
 			ServiceAccount: *snapshotSvcAccPtr,
 			Src:            true,
 		}
-		mfest = reg.Manifest{
-			Registries: []reg.RegistryContext{
-				*srcRegistry,
+		mfests = []reg.Manifest{
+			{
+				Registries: []reg.RegistryContext{
+					*srcRegistry,
+				},
+				Images: []reg.Image{},
 			},
-
-			Images: []reg.Image{},
 		}
 	} else {
-		if *manifestPtr == "" {
-			klog.Fatal(fmt.Errorf("-manifest=... flag is required"))
+		if *manifestPtr == "" && *manifestDirPtr == "" {
+			klog.Fatal(fmt.Errorf("-manifest=... or -manifestDir=... flag is required"))
 		}
+	}
+
+	if *manifestPtr != "" {
 		mfest, err = reg.ParseManifestFromFile(*manifestPtr)
 		if err != nil {
 			klog.Fatal(err)
 		}
-	}
+		mfests = append(mfests, mfest)
+		for _, registry := range mfest.Registries {
+			mi[registry.Name] = nil
+		}
+		sc, err = reg.MakeSyncContext(
+			mfests,
+			*verbosityPtr,
+			*threadsPtr,
+			*dryRunPtr,
+			!noSvcAcc)
+		if err != nil {
+			klog.Fatal(err)
+		}
+	} else if *manifestDirPtr != "" {
+		mfests, err = reg.ParseManifestsFromDir(*manifestDirPtr)
+		if err != nil {
+			klog.Exitln(err)
+		}
 
-	sc, err := reg.MakeSyncContext(
-		mfest.Filepath(),
-		mfest,
-		*verbosityPtr,
-		*threadsPtr,
-		*dryRunPtr,
-		!noSvcAcc)
-	if err != nil {
-		klog.Fatal(err)
+		err = reg.ValidateManifestsFromDir(mfests)
```

Comment:
Would it be easier if ParseManifestsFromDir validates, given that ParseManifestsFromFile does?  (or vice-versa)

Reply:
Main reason I split it off like this was for trying to keep the function sizes reasonably small. ParseManifestsFromDir is already quite large. On the other hand I don't want to just rely on ParseManifestFromFile (I think think is what you meant? there is no ParseManifestsFromFile) because 2 independently-valid manifests != 1 valid manifest folder of those same 2 manifests.